azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -417,21 +447,39 @@ def supported(self):
             return self._cgroups_supported
 
         def enabled(self):
-            return self._cgroups_enabled
+            return self._agent_cgroups_enabled | self._extension_cgroups_enabled
+
+        def agent_enabled(self):
+            return self._agent_cgroups_enabled
+
+        def extension_enabled(self):
+            return self._extension_cgroups_enabled
 
         def enable(self):
             if not self.supported():
                 raise CGroupsException("Attempted to enable cgroups, but they are not supported on the current platform")
-            self._cgroups_enabled = True
+            self._agent_cgroups_enabled = True
+            self._extension_cgroups_enabled = True
             self.__set_cpu_quota(conf.get_agent_cpu_quota())
 
-        def disable(self, reason):
-            self._cgroups_enabled = False
+        def disable(self, reason, telemetry_name = None):
```

Comment:
taking a telemetry name as parameter gives the impression that one can pass any name, for example the name of an extension, but that is not the case.

an enum (all, agent, extensions) may be more appropriate

Reply:
yes, Idea is to get the extension name(same name defined as telemetry name) if ever want to disable that particular extension when we decide to that way in future.

Reply:
For now that is not the case, we can keep as an enums